test/common/ipc_prov_consumer.h

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <umf/memory_provider.h>
+
+#define INET_ADDR "127.0.0.1"
+#define MSG_SIZE 256
+#define RECV_BUFF_SIZE 1024
+
+// consumer's response message
+#define CONSUMER_MSG                                                           \
+    "This is the consumer. I just wrote a new number directly into your "      \
+    "shared memory!"
+
+/*
+Generally communication between the producer and the consumer looks like:
+- Consumer starts
+- Consumer creates a socket
+- Consumer listens for incoming connections
+- Producer starts
+- Producer's shared memory contains a number: N
+- Producer gets the IPC handle
+- Producer creates a socket
+- Producer connects to the consumer
+- Consumer connects at IP 127.0.0.1 and a port to the producer
+- Producer sends the IPC handle to the consumer
+- Consumer receives the IPC handle from the producer
+- Consumer opens the IPC handle received from the producer
+- Consumer reads the number from the producer's shared memory: N
+- Consumer writes a new number directly to the producer's shared memory: N/2
+- Consumer sends a response message to the producer
+- Consumer closes the IPC handle received from the producer
+- Producer receives the response from the consumer: "This is the consumer. I just wrote a new number directly into your shared memory!"
+- Producer verifies the consumer wrote the correct value (the old one / 2) to the producer's shared memory: N/2
+- Producer puts the IPC handle
+- Consumer shuts down
+- Producer shuts down
+*/
+
+int consumer_connect(int port) {
```

Comment:
The header files should NOT contain definitions (implementations) of functions, but they should contain declarations only. It is not a header file now - it is a source file.

Reply:
BTW: changes that actually do not change a code, but only moves it to another file should be placed in a separate PR. It would make a review of such patches much easier.

Reply:
> The header files should not contain definitions (implementations) of functions, but they should contain declarations only. It is not a header file now - it is a source file.

In general, I agree with that. But in the case of tests, we have one source file per test binary. Do you still suggest to move generic functionality from headers to separate src?

Reply:
1) First of all, this change has nothing to do with the topic of this PR, so it is should be moved to a separate PR, where we can continue this discussion.
2) I still think that each header should be a correct header file, that can be included multiple times and that does not contain any source code except of inline functions.

Reply:
@vinser52 Please review and use: https://github.com/oneapi-src/unified-memory-framework/pull/574

Reply:
and https://github.com/oneapi-src/unified-memory-framework/pull/576 ...

Reply:
Fixed in https://github.com/oneapi-src/unified-memory-framework/pull/574